src/BlazorWebView/src/core/MauiDispatcher.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.Maui.Controls;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.WebView.Maui
+{
+	internal class MauiDispatcher : Dispatcher
+	{
+		public static Dispatcher Instance { get; } = new MauiDispatcher();
+
+		private MauiDispatcher()
+		{
+		}
+
+#pragma warning disable CA1416 // Validate platform compatibility
+		public override bool CheckAccess()
+		{
+			return !Device.IsInvokeRequired;
+		}
+
+		public override Task InvokeAsync(Action workItem)
+		{
+			return Device.InvokeOnMainThreadAsync(workItem);
+		}
+
+		public override Task InvokeAsync(Func<Task> workItem)
+		{
+			return Device.InvokeOnMainThreadAsync(workItem);
+		}
+
+		public override Task<TResult> InvokeAsync<TResult>(Func<TResult> workItem)
+		{
+			return Device.InvokeOnMainThreadAsync(workItem);
+		}
+
+		public override Task<TResult> InvokeAsync<TResult>(Func<Task<TResult>> workItem)
+		{
+			return Device.InvokeOnMainThreadAsync(workItem);
+		}
+#pragma warning restore CA1416 // Validate platform compatibility
```

Comment:
nit: in these we might be able to check if we are already on the UI thread and invoke the function directly if we do

Reply:
Will make a note.